pyarbtools/instruments.py

```diff
@@ -1384,6 +1392,138 @@ def bin_pdw_builder(self, operation=0, freq=1e9, phase=0, startTimeSec=0, width=
+
+            Args:
+                sizeOfPaddingAndHeaderInBytes (int): Total size of resulting padding
+                    binary block and header combined.
+            Returns:
+                binary block containing padding header and padded data
+        """
+
+        paddingHeaderSize = 16
+        paddingFillerSize = sizeOfPaddingAndHeaderInBytes - paddingHeaderSize
+
+        padBlockId = (1).to_bytes(4, byteorder='little')
+        res3 = (0).to_bytes(4, byteorder='little')
+        size = (paddingFillerSize).to_bytes(8, byteorder='little')
+        # Padding Header Above = 16 bytes
+
+        # X bytes of padding required to ensure PDW stream contents
+        # (not PDW header) starts @ byte 4097 or (multiple of 4096)+1
+        padData = (0).to_bytes(paddingFillerSize, byteorder='little')
+        padding = [padBlockId, res3, size, padData]
+
+        return padding
+
+
+    def bin_freqPhaseCodingSingleEntry(self, onOffState=0, numBitsPerSubpulse=1, codingType=0,
+                                       stateMapping=[0,180], hexPatternString="E2",
+                                       comment="default Comment"):
+        """Creates a single entry binary frequency and phase coding block
+         for analog UXG streaming.  This is only part of a full frequency and phase coding
+         block with multiple entries for each pattern to be streamed to UXG.
+             Args:
+                 onOffState (int): Activation state for current FPC entry
+                 numBitsPerSubpulse (int): = number of bits per subpulse.  E.g. For BPSK, this is 1
+                 codingType (int): 0=phase coding, 1= frequency coding, 2 = both phase and frequency coding
+                 stateMapping (double array): 2^numBitsPerSubpulse entries of phase / freq states
+                 hexPatternString (string):  Hex values to encode in FPC table e.g. "A2F4" multiple of 2 in length
+                 comment (string): FPC entry name
+
+             Returns:
+                 binary array containing bytes for a single frequency phase entry
+
+             TODO - Combination of simultaneous phase and frequency modulation not yet implemented
+        """
+        if ((len(hexPatternString) % 2) != 0):
+            raise Exception('Hex pattern length must be a multiple of 2: Length is ' + str(len(hexPatternString)))
+
+        hexPatternBytes = bytearray.fromhex(hexPatternString)
+        numBitsInPattern = 8 * len(hexPatternBytes)
+
+        if (codingType !=0 and codingType !=1):
+            raise Exception('Only phase and frequency coding via streaming has been implemented in this example')
+        if (numBitsPerSubpulse != 1):
+            raise Exception('Only one bit per subpulse has been implemented in this example')
+        if (len(hexPatternBytes) > 8192):
+            raise Exception('Pattern must be less than 8192 bytes')
+        if (len(comment) > 60):
+            raise Exception('Comment must be less than 60 characters long')
```

Comment:
Agreed.  Should each line be changed from "raise Exception" to "raise error.UXGError"?  How would I correct my committed fork?  Re-commit and submit another pull request?

Reply:
Yes, that's the correct fix. Based on my reading, you should be able to make changes and then commit them and it will automatically update the pull request.